### PR TITLE
Add the spack tutorial environment as a cloud pipeline stack

### PR DESCRIPTION
**Pull request by tldahlgren**
This PR adds the Spack tutorial (environment) as a cloud pipeline stack.  

The goal is to ensure Spack is building a cache of binaries for the tutorial to simplify the tutorial preparation process.

### CODE REVIEW

File: share/spack/gitlab/cloud_pipelines/stacks/tutorial/spack.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  - $gcc_old_packages
+  - $clang_packages
+  - $gcc_spack_built_packages
+
+  mirrors:
+    mirror: 's3://spack-binaries-develop/tutorial'
+
+  gitlab-ci:
+    script:
+      - . "./spack/share/spack/setup-env.sh"
```
Comment (scottwittenburg) [inline share/spack/gitlab/cloud_pipelines/stacks/tutorial/spack.yaml]: Copy-pasted from the spack-tutorial-container repo?  In this project, CI is run inside the spack directory, so this should be `./share/spack/setup-env.sh` instead.

Comment (tldahlgren) [reply share/spack/gitlab/cloud_pipelines/stacks/tutorial/spack.yaml]: Thanks for catching that!

Comment (scottwittenburg) [reply share/spack/gitlab/cloud_pipelines/stacks/tutorial/spack.yaml]: Actually gitlab caught it 😆

File: share/spack/gitlab/cloud_pipelines/stacks/tutorial/spack.yaml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  - $clang_packages
+  - $gcc_spack_built_packages
+
+  mirrors:
+    mirror: 's3://spack-binaries-develop/tutorial'
+
+  gitlab-ci:
+    script:
+      - . "./share/spack/setup-env.sh"
+      - spack --version
```
Comment (scottwittenburg) [inline share/spack/gitlab/cloud_pipelines/stacks/tutorial/spack.yaml:61]: I think we might need a `spack compiler find` here.  And I'm not sure, but we might need to run the pipeline generation on the same docker image as these build jobs, but let's hold off on that and see if the compiler find does what we need first.

File: share/spack/gitlab/cloud_pipelines/stacks/tutorial/spack.yaml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        - tcl ^zlib@1.2.8 cppflags=-O3
+        - hdf5
+        - hdf5~mpi
+        - hdf5+hl+mpi ^mpich
+        - trilinos
+        - trilinos +hdf5 ^hdf5+hl+mpi ^mpich
+        - gcc@8.4.0
+        - mpileaks
+        - lmod
+        - macsio@1.1+scr^scr@2.0.0~fortran^silo~fortran^hdf5~fortran
```
Comment (scottwittenburg) [inline share/spack/gitlab/cloud_pipelines/stacks/tutorial/spack.yaml:32]: Is `scr@2.0.0` too old to build anymore?  Seems spack didn't want to fetch it.

Comment (tldahlgren) [reply share/spack/gitlab/cloud_pipelines/stacks/tutorial/spack.yaml:32]: Waiting on https://github.com/spack/spack/pull/27178 to be merged.

**Comment by scottwittenburg** [comment]
@spackbot run pipeline

**Comment by tldahlgren** [comment]
Aw.  Now we're having missing hash issues with the E4S stack for the `hpx` package:

```
spack.error.SpackError: No installed spec matches the hash: '6l4tny4cwy5xon3umjbh6fisvshal354'
```

**Comment by scottwittenburg** [comment]
> missing hash issues with the E4S stack for the `hpx` package

Can you rebase?  That was fixed with #27215.

**Review by scottwittenburg — APPROVED** [review]

**Pull request merged**